test/hummingbot/strategy/avellaneda_market_making/test_avellaneda_market_making.py

```diff
@@ -0,0 +1,1145 @@
+        self.assertEqual(expected_output, self.strategy.create_proposal_based_on_order_override())
+
+        order_override = {
+            "order_1": ["sell", 2.5, 100],
+            "order_2": ["buy", 0.5, 100]
+        }
+
+        # Re-configure strategy with order_ride configurations
+        self.strategy.order_override = order_override
+
+        expected_proposal = (list(), list())
+        for order in order_override.values():
+            list_to_append = expected_proposal[0] if order[0] == "buy" else expected_proposal[1]
+            if "buy" == order[0]:
+                price = self.strategy.get_price() * (Decimal("1") - Decimal(str(order[1])) / Decimal("100"))
+            else:
+                price = self.strategy.get_price() * (Decimal("1") + Decimal(str(order[1])) / Decimal("100"))
+
+            price = self.market.quantize_order_price(self.trading_pair, price)
+            size = self.market.quantize_order_amount(self.trading_pair, Decimal(str(order[2])))
+
+            list_to_append.append(PriceSize(price, size))
+
+        self.assertEqual(str(expected_proposal), str(self.strategy.create_proposal_based_on_order_override()))
+
+    def test_get_logspaced_level_spreads(self):
+        # Re-initialize strategy with order_level configurations
+        self.strategy = AvellanedaMarketMakingStrategy(
+            market_info=self.market_info,
+            order_amount=self.order_amount,
+            order_levels=2,
+        )
+        self.strategy.start(self.clock, self.start_timestamp)
+
+        # Simulate low volatility.
+        # Note: bid/ask_level_spreads Requires volatility, optimal_bid, optimal_ask to be defined
+        self.simulate_low_volatility(self.strategy)
+
+        self.strategy.recalculate_parameters()
+        self.strategy.calculate_reserved_price_and_optimal_spread()
+
+        # Calculation for expected bid/ask_level_spreads
+        reference_price = self.strategy.get_price()
+        _, max_spread = self.strategy._get_min_and_max_spread()
+        optimal_ask_spread = self.strategy.optimal_ask - reference_price
+        optimal_bid_spread = reference_price - self.strategy.optimal_bid
+        expected_bid_spreads = np.logspace(0, np.log(float(max_spread - optimal_bid_spread) + 1), base=np.e,
+                                           num=2) - 1
+        expected_ask_spreads = np.logspace(0, np.log(float(max_spread - optimal_ask_spread) + 1), base=np.e,
+                                           num=2) - 1
+
+        bid_level_spreads, ask_level_spreads = self.strategy._get_logspaced_level_spreads()
+        for i, spread in enumerate(bid_level_spreads):
+            self.assertAlmostEqual(expected_bid_spreads[i], spread, 1)
+
+        for i, spread in enumerate(ask_level_spreads):
+            self.assertAlmostEqual(expected_ask_spreads[i], spread, 1)
+
+        # Simulate high volatility. TODO: Find a better max_spread parameter to better illustrate bid levels
```

Comment:
Is this TODO still pending?

Reply:
See comment [here](https://github.com/CoinAlpha/hummingbot/pull/3483#discussion_r655374449).